File: coreServer/src/main/java/com/example/coreServer/dto/CreateChatRequest.java
package com.example.coreServer.dto;

import com.example.coreServer.model.ChatType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateChatRequest {

    /**
     * DIRECT / GROUP / CHANNEL
     */
    private ChatType type;

    /**
     * Заголовок чата (для DIRECT можно не заполнять или генерировать на бэке).
     */
    private String title;

    /**
     * Владелец чата (создатель).
     */
    private Long ownerId;

    /**
     * Участники чата (id пользователей).
     * Для DIRECT обычно 2 id: ownerId и собеседник.
     */
    private List<Long> memberIds;
}

File: CS Middleware/Shared/DTOs/UserDTOs/CreateChatRequest.cs
using System;

namespace DTOs.UserDTOs;

public class CreateChatRequest {
    public ChatType Type { get; set; }
    public required String Title { get; set; }
    public long OwnerId { get; set; }
    public List<long> MemberIds { get; set; } = [];
}

public enum ChatType {
    DIRECT,
    GROUP,
    CHANNEL,
}
